cmd/insolard/main.go

```diff
@@ -69,60 +146,91 @@ func runInsolardServer(configPath string, genesisConfigPath string, genesisOnly
 		log.Fatal(errors.Wrap(err, "failed to get API info response"))
 	}
 
-	switch role {
-	case insolar.StaticRoleHeavyMaterial:
-		states, _ := initStates(configPath, genesisConfigPath)
-		s := server.NewHeavyServer(
-			configPath,
-			genesisConfigPath,
-			genesis.Options{
-				States:       states,
-				ParentDomain: application.GenesisNameRootDomain,
-			},
-			genesisOnly,
-			apiOptions,
-		)
-		s.Serve()
-	case insolar.StaticRoleLightMaterial:
-		s := server.NewLightServer(
-			configPath,
-			apiOptions,
-		)
-		s.Serve()
-	case insolar.StaticRoleVirtual:
-		builtinContracts := builtin.BuiltinContracts{
-			CodeRegistry:         appbuiltin.InitializeContractMethods(),
-			CodeRefRegistry:      appbuiltin.InitializeCodeRefs(),
-			CodeDescriptors:      appbuiltin.InitializeCodeDescriptors(),
-			PrototypeDescriptors: appbuiltin.InitializePrototypeDescriptors(),
-		}
-		s := server.NewVirtualServer(
-			configPath,
-			builtinContracts,
-			apiOptions,
-		)
-		s.Serve()
+	builtinContracts := builtin.BuiltinContracts{
```

Comment:
why??? sorry, but isn't it a part of deep virtual initialization?

Reply:
futher discussion needed